### PR TITLE
[daemon] Skip ghost instances in JSON DB

### PR DESCRIPTION
**Pull request by ricab**
Skip entries where all members are 0/empty, which we know are ghosts of
purged instances. They get removed in the following DB update.

Addresses #1935.

### CODE REVIEW

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -311,6 +311,14 @@ std::unordered_map<std::string, mp::VMSpecs> load_db(const mp::Path& data_path,
         auto deleted = record["deleted"].toBool();
         auto metadata = record["metadata"].toObject();
 
+        if (!num_cores && !state && !deleted && ssh_username.empty() && metadata.isEmpty() &&
+            !mp::MemorySize{mem_size}.in_bytes() && !mp::MemorySize{disk_space}.in_bytes())
+        {
+            mpl::log(multipass::logging::Level::warning, category,
```
Comment (Saviq) [inline src/daemon/daemon.cpp:317]:
```suggestion
            mpl::log(mpl::Level::warning, category,
```

Comment (ricab) [reply src/daemon/daemon.cpp:317]: Oh right, definitely.

**Comment by ricab** [comment]
I have confirmed this can now handle the ghosts on the mac.

**Comment by Saviq** [comment]
macOS build available: [multipass-1.7.0-dev.572.pr1944+g31e64a91.mac-Darwin.pkg](https://multipass-ci.s3.amazonaws.com/pr1944/multipass-1.7.0-dev.572.pr1944%2Bg31e64a91.mac-Darwin.pkg)

**Review by Saviq — APPROVED** [review]
Yup!

```json
$ sudo cat /Volumes/Macintosh\ HD/multipass/data/multipassd-vm-instances.json
{
    "athletic-rhinoceros": {
        "deleted": false,
        "disk_space": "0",
        "mac_addr": "",
        "mem_size": "0",
        "metadata": {
        },
        "mounts": [
        ],
        "num_cores": 0,
        "ssh_username": "",
        "state": 0
    }
}

$ multipass version
multipass  1.7.0-dev.572.pr1944+g31e64a91.mac
multipassd 1.7.0-dev.572.pr1944+g31e64a91.mac

$ multipass launch 
Launched: evocative-genet                   
                                    
$ sudo cat /Volumes/Macintosh\ HD/multipass/data/multipassd-vm-instances.json
{
    "evocative-genet": {
        "deleted": false,
        "disk_space": "5368709120",
        "extra_interfaces": [
        ],
        "mac_addr": "52:54:00:93:99:88",
        "mem_size": "1073741824",
        "metadata": {
        },
        "mounts": [
        ],
        "num_cores": 1,
        "ssh_username": "ubuntu",
        "state": 4
    }
}

$ multipass delete -p --all

$ sudo cat /Volumes/Macintosh\ HD/multipass/data/multipassd-vm-instances.json
{
}
```

bors r+

**Pull request merged**